argo-workflows/okd-workflows/30-tpl-build-memoize.yaml

```diff
@@ -0,0 +1,136 @@
+kind: WorkflowTemplate
+apiVersion: argoproj.io/v1alpha1
+metadata:
+  name: wftpl-build-memoize
+spec:
+  templates:
+    - name: multiarch-build # formerly build-multiarch-workflow
+      inputs:
+        parameters:
+          - name: architectures
+          - name: build-config-name
+          - name: branch
+            default: ""
+          - name: repo-url
+            default: ""
+      #retryStrategy:
+      #  backoff:
+      #    duration: 1m
+      #    factor: "2"
+      #    maxDuration: 60m
+      #  limit: "3"
+      #  retryPolicy: Always
+      steps:
+        - - name: generate-memoization-key
+            arguments:
+              parameters:
+                - name: build-config-name
+                  value: "{{inputs.parameters.build-config-name}}"
+            template: prepare-memoization-key
+        - - name: multiarch-build
+            template: multiarch-builds
+            arguments:
+              parameters:
+                - name: architectures
+                  value: "{{inputs.parameters.architectures}}"
+                - name: branch
+                  value: "{{inputs.parameters.branch}}"
+                - name: build-config-name
+                  value: "{{inputs.parameters.build-config-name}}"
+                - name: memoization-key
+                  value: "{{steps.generate-memoization-key.outputs.parameters.memoization-key}}"
+                - name: repo-url
+                  value: "{{inputs.parameters.repo-url}}"
+    - name: prepare-memoization-key
+      volumes:
+        - name: tmp
+          emptyDir: {}
+      inputs:
+        parameters:
+          - name: build-config-name
+      outputs:
+        parameters:
+          - name: memoization-params
+            valueFrom:
+              path: /tmp/volume/memoization-params
+          - name: memoization-key
+            valueFrom:
+              path: /tmp/volume/memoization-key
+      containerSet:
+        containers:
+          - name: prepare-memoization-params
+            image: image-registry.openshift-image-registry.svc:5000/openshift/cli
+            command: [bash, -c]
+            args:
+              - |
+                set -ex
+                echo "Preparing memoization params"
+                git_uri=$(oc get bc {{inputs.parameters.build-config-name}} -o jsonpath='{.spec.source.git.uri}')
+                git_revision=$(oc get bc {{inputs.parameters.build-config-name}} -o jsonpath='{.spec.source.git.ref}')
```

Comment:
how do we deal with errors in oc commands/timeouts etc..?

Reply:
Hi @Prashanth684 it would fail for such issues. However, we never wait for long-runing executions (as in `oc wait bc`) to conclude because Argo Workflow has its own ways for doing it.